applications/GeoMechanicsApplication/tests/cpp_tests/custom_processes/test_fix_water_pressures_above_phreatic_line.cpp

```diff
@@ -0,0 +1,244 @@
+        FixWaterPressuresAbovePhreaticLineProcess(r_model_part, test_parameters), expected_error_message);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(TestFixWaterPressureAbovePhreaticLine_FixesAllWaterPressuresToZeroWhenAllNodesAbovePhreaticLine,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    constexpr double y_node_1     = 0.0;
+    constexpr double y_node_2     = -1.0;
+    auto             model        = Model{};
+    auto&            r_model_part = CreateModelPartWithTwoNodesAtHeights(model, y_node_1, y_node_2);
+    auto             test_parameters = CreateParametersWithConstantPhreaticLineAtHeight(-2.0);
+    r_model_part.GetNode(1).FastGetSolutionStepValue(WATER_PRESSURE) = 1.0;
+    r_model_part.GetNode(2).FastGetSolutionStepValue(WATER_PRESSURE) = 2.0;
+    FixWaterPressuresAbovePhreaticLineProcess process(r_model_part, test_parameters);
+
+    // Act
+    process.ExecuteInitializeSolutionStep();
+
+    // Assert
+    for (auto& r_node : r_model_part.Nodes()) {
+        EXPECT_DOUBLE_EQ(r_node.FastGetSolutionStepValue(WATER_PRESSURE), 0.0);
+        EXPECT_TRUE(r_node.IsFixed(WATER_PRESSURE));
+    }
+}
+
+KRATOS_TEST_CASE_IN_SUITE(TestFixWaterPressureAbovePhreaticLine_DoesNothingWhenAllNodesBelowPhreaticLine,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    constexpr double y_node_1     = 0.0;
+    constexpr double y_node_2     = -1.0;
+    auto             model        = Model{};
+    auto&            r_model_part = CreateModelPartWithTwoNodesAtHeights(model, y_node_1, y_node_2);
+    r_model_part.GetNode(1).FastGetSolutionStepValue(WATER_PRESSURE) = 1.0;
+    r_model_part.GetNode(2).FastGetSolutionStepValue(WATER_PRESSURE) = 2.0;
+
+    auto test_parameters = CreateParametersWithConstantPhreaticLineAtHeight(1.0);
+    FixWaterPressuresAbovePhreaticLineProcess process(r_model_part, test_parameters);
+
+    // Act
+    process.ExecuteInitializeSolutionStep();
+
+    // Assert
+    EXPECT_DOUBLE_EQ(r_model_part.GetNode(1).FastGetSolutionStepValue(WATER_PRESSURE), 1.0);
+    EXPECT_FALSE(r_model_part.GetNode(1).IsFixed(WATER_PRESSURE));
+    EXPECT_DOUBLE_EQ(r_model_part.GetNode(2).FastGetSolutionStepValue(WATER_PRESSURE), 2.0);
+    EXPECT_FALSE(r_model_part.GetNode(2).IsFixed(WATER_PRESSURE));
+}
+
+KRATOS_TEST_CASE_IN_SUITE(TestFixWaterPressureAbovePhreaticLine_OnlyFixesNodesAbovePhreaticLine,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    constexpr double y_node_1     = 0.0;
+    constexpr double y_node_2     = -1.0;
+    auto             model        = Model{};
+    auto&            r_model_part = CreateModelPartWithTwoNodesAtHeights(model, y_node_1, y_node_2);
+
+    r_model_part.GetNode(1).FastGetSolutionStepValue(WATER_PRESSURE) = 1.0;
+    r_model_part.GetNode(2).FastGetSolutionStepValue(WATER_PRESSURE) = 2.0;
+
+    auto test_parameters = CreateParametersWithConstantPhreaticLineAtHeight(-0.5);
```

Comment:
And this one, can we make this variable `const`?
```suggestion
    const auto test_parameters = CreateParametersWithConstantPhreaticLineAtHeight(-0.5);
```

Reply:
Done